cli/js/diagnostics_util.ts

```diff
@@ -10,6 +10,79 @@ import {
   DiagnosticItem,
 } from "./diagnostics.ts";
 
+const unstableDenoGlobalProperties = [
+  "umask",
+  "linkSync",
+  "link",
+  "symlinkSync",
+  "symlink",
+  "DirKind",
+  "dir",
+  "loadavg",
+  "osRelease",
+  "openPlugin",
+  "DiagnosticCategory",
+  "DiagnosticMessageChain",
+  "DiagnosticItem",
+  "Diagnostic",
+  "formatDiagnostics",
+  "CompilerOptions",
+  "TranspileOnlyResult",
+  "transpileOnly",
+  "compile",
+  "bundle",
+  "Location",
+  "applySourceMap",
+  "LinuxSignal",
+  "MacOSSignal",
+  "Signal",
+  "SignalStream",
+  "signal",
+  "signals",
+  "setRaw",
+  "utimeSync",
+  "utime",
+  "ShutdownMode",
+  "shutdown",
+  "DatagramConn",
+  "UnixListenOptions",
+  "listen",
+  "listenDatagram",
+  "UnixConnectOptions",
+  "connect",
+  "StartTlsOptions",
+  "startTls",
+  "kill",
+  "PermissionName",
+  "PermissionState",
+  "RunPermissionDescriptor",
+  "ReadPermissionDescriptor",
+  "WritePermissionDescriptor",
+  "NetPermissionDescriptor",
+  "EnvPermissionDescriptor",
+  "PluginPermissionDescriptor",
+  "HrtimePermissionDescriptor",
+  "PermissionDescriptor",
+  "Permissions",
+  "PermissionStatus",
+  "hostname",
+];
+
+function transformMessageText(messageText: string, code: number): string {
+  if (code === 2339) {
+    const property = messageText
+      .replace(/^Property '/, "")
+      .replace(/' does not exist on type 'typeof Deno'\.$/, "");
+    if (
+      messageText.endsWith("on type 'typeof Deno'.") &&
+      unstableDenoGlobalProperties.includes(property)
+    ) {
+      return messageText + " Did you forget to run with the '--unstable' flag?";
```

Comment:
`<property> is an unstable API; did you forget to run with the '--unstable' flag?`